src/index/sparse/sparse_inverted_index.h

```diff
@@ -516,135 +536,128 @@ class InvertedIndex : public BaseInvertedIndex<T> {
 
-    // find the top-k candidates using brute force search, k as specified by the capacity of the heap.
-    // any value in q_vec that is smaller than q_threshold and any value with dimension >= n_cols() will be ignored.
-    // TODO: may switch to row-wise brute force if filter rate is high. Benchmark needed.
-    void
-    search_brute_force(const SparseRow<T>& q_vec, T q_threshold, MaxMinHeap<T>& heap, const BitsetView& bitset,
-                       const DocValueComputer<T>& computer) const {
-        auto scores = compute_all_distances(q_vec, q_threshold, computer);
-        for (size_t i = 0; i < n_rows_internal(); ++i) {
-            if ((bitset.empty() || !bitset.test(i)) && scores[i] != 0) {
-                heap.push(i, scores[i]);
-            }
-        }
-    }
-
-    // LUT supports size() and operator[] which returns an SparseIdVal.
-    template <typename LUT>
+    template <typename DocIdFilter>
     struct Cursor {
      public:
-        Cursor(const LUT& lut, size_t num_vec, float max_score, float q_value, const BitsetView bitset)
-            : lut_(lut),
-              lut_size_(lut.size()),
+        Cursor(const Vector<table_t>& plist_ids, const Vector<QType>& plist_vals, size_t num_vec, float max_score,
+               float q_value, const DocIdFilter filter)
+            : plist_ids_(plist_ids),
+              plist_vals_(plist_vals),
+              plist_size_(plist_ids.size()),
               total_num_vec_(num_vec),
               max_score_(max_score),
               q_value_(q_value),
-              bitset_(bitset) {
-            while (loc_ < lut_size_ && !bitset_.empty() && bitset_.test(lut_[loc_].id)) {
-                loc_++;
-            }
+              filter_(filter) {
+            skip_filtered_ids();
             update_cur_vec_id();
         }
         Cursor(const Cursor& rhs) = delete;
+        Cursor(Cursor&& rhs) noexcept = default;
 
         void
         next() {
-            next_internal();
+            ++loc_;
+            skip_filtered_ids();
             update_cur_vec_id();
         }
 
-        // advance loc until cur_vec_id_ >= vec_id
         void
         seek(table_t vec_id) {
-            while (loc_ < lut_size_ && lut_[loc_].id < vec_id) {
-                next_internal();
+            while (loc_ < plist_size_ && plist_ids_[loc_] < vec_id) {
+                ++loc_;
             }
+            skip_filtered_ids();
             update_cur_vec_id();
         }
 
-        T
+        QType
         cur_vec_val() const {
-            return lut_[loc_].val;
+            return plist_vals_[loc_];
         }
 
-        const LUT& lut_;
-        const size_t lut_size_;
+        const Vector<table_t>& plist_ids_;
+        const Vector<QType>& plist_vals_;
+        const size_t plist_size_;
         size_t loc_ = 0;
         size_t total_num_vec_ = 0;
         float max_score_ = 0.0f;
         float q_value_ = 0.0f;
-        const BitsetView bitset_;
+        const DocIdFilter filter_;
```

Comment:
what's the reason for making this `const`?

Reply:
`class DocIdFilterByVector` is actually another form of `BitsetView`, so the template typename `DocIdFilter` could be `BitsetView` or `DocIdFilterByVector`.

`knowhere::sparse::InvertedIndex::Search()` takes a `const BitsetView& bitset` as an argument. And `Cursor` will use this argument to iterate through the entire posting list.

To keep consistency, the type of `DocIdFilter` also needs to be a `const` type.